core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala

```diff
@@ -237,214 +238,298 @@ class DeleteTopicTest extends QuorumTestHarness {
     ensureControllerExists()
     // wait until a new controller to show up
     TestUtils.waitUntilTrue(() => {
-      val (newController, newControllerId) = getController()
+      val (_, newControllerId) = getController()
       newControllerId != previousControllerId
     }, "The new controller should not have the failed controller id")
 
     // bring back the failed brokers
     follower.startup()
     controller.startup()
-    TestUtils.verifyTopicDeletion(zkClient, topic, 2, servers)
-    adminClient.close()
+    TestUtils.verifyTopicDeletion(zkClientOrNull, topic, 2, brokersWithReplicas)
+  }
+
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("kraft"))
+  def testIncreasePartitionCountDuringDeleteTopicKRaftMode(quorum: String): Unit = {
+    val topicPartition = new TopicPartition("test", 0)
+    val topic = topicPartition.topic
+    createTestTopicAndCluster(topic, numConfigs = 4)
+    val brokersWithReplicas = brokers.filter(s => expectedReplicaAssignment(0).contains(s.config.brokerId))
+    // shutdown a broker to make sure the following topic deletion will be suspended
+    val partition = getTopicPartitionInfo(adminClient, topic, 0)
+    assertTrue(partition.isPresent, "Partition [test,0] should exist")
+    val leaderNode = partition.get().leader()
+    assertFalse(leaderNode.isEmpty, "Leader should exist for partition [test,0]")
+    val follower = brokersWithReplicas.filter(s => s.config.brokerId != leaderNode.id()).last
+    follower.shutdown()
+    // start topic deletion
+    faultHandler.setIgnore(true)
+    adminClient.deleteTopics(Collections.singleton(topic))
+
+    // make sure deletion of all of the topic's replicas have been tried
+    val aliveBrokers = brokers.filter(b => b != follower)
+    TestUtils.waitForAllPartitionsMetadata(aliveBrokers, topic, 0)
+    TestUtils.waitForAllPartitionsMetadata(Seq(follower), topic, 1)
+    // increase the partition count for topic
+    try {
+      adminClient.createPartitions(Map(topic -> NewPartitions.increaseTo(2)).asJava).all().get()
+    } catch {
+      case _: ExecutionException =>
```

Comment:
This is another case where the different behavior may be causing some confusion. This fails in zookeeper mode probably because the topic is marked for deletion as the test expects. However, in kraft, it probably fails because the topic doesn't exist. We could try to create the topic instead to see the difference in behavior.